tests/system/src/test/java/test/javafx/embed/swing/SwingNodeContentMemoryLeakTest.java

```diff
@@ -0,0 +1,137 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.embed.swing;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import javafx.application.Application;
+import javafx.embed.swing.SwingNode;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import javafx.stage.Stage;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.sun.javafx.PlatformUtil;
+
+import test.util.Util;
+
+public class SwingNodeContentMemoryLeakTest {
+
+    static CountDownLatch launchLatch = new CountDownLatch(1);
+
+    //Keep week references to all panels that we've ever generated to see if any
+    //of them get collected.
+    private Collection<WeakReference<JPanel>> panels = new CopyOnWriteArrayList<>();
+    private int count = 0;
+    private int fail = 0;
+    private SwingNode node;
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, 50, MyApp.class);
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void testSwingNodeContentMemoryLeak() throws InterruptedException,
+                                                        InvocationTargetException {
+        Util.runAndWait(() -> {
+            node = new SwingNode();
+            Pane root = new Pane();
+            root.getChildren().add(node);
+
+            Stage stage = new Stage();
+            Scene scene = new Scene(root, 150, 100);
+            stage.setScene(scene);
+        });
+
+        //Kick off a thread that repeatedly creates new JPanels and resets the swing node's content
+        new Thread(() -> {
+            while(count < 50) {
+                //Lets throw in a little sleep so we can read the output
+                try {
+                    Thread.sleep(100);
```

Comment:
they become predictable when the seed is known.

Reply:
Yes, but then you would need to modify the test to rerun it with the same seed as the time it failed. If we had a utility that made this easy, then maybe, but I'd still be skeptical. I am not convinced that whatever benefit we might get is worth the effort or the unpredictability.

Reply:
the main benefit is that every new run effectively extends the test coverage.
however, in this particular case, the test won't be reproducible because it depends on many other factors like the platform load.